scripts/upload-daemon/Main.hs

```diff
@@ -31,21 +32,42 @@ data StatisticsHandlers
 
 type UploadTarget = String
 
--- | First element of a list that is not Nothing
-firstJust :: [Maybe a] -> Maybe a
-firstJust = join . listToMaybe . Prelude.filter isJust
-
+data UploadOptions = UploadOptions
+  { port :: Int
+  , prometheusPort :: Int
+  , uploadTarget :: UploadTarget
+  , nrWorkers :: Int
+  }
 
--- | Upload target is taken either from first argument or from UPLOAD_TARGET env variable
-getUploadTarget :: IO (Maybe UploadTarget)
-getUploadTarget = firstJust <$> sequence
-  [ listToMaybe <$> getArgs
-  , optional $ getEnv "UPLOAD_TARGET"
-  ]
+uploadOptions :: Parser UploadOptions
+uploadOptions = UploadOptions
+  <$> option auto
+  ( long "port"
+    <> short 'p'
+    <> metavar "PORT"
+    <> value 8080
+    <> showDefault
+    <> help "Daemon listening port" )
+  <*> option auto
+  ( long "stat-port"
+    <> short 's'
+    <> metavar "SPORT"
+    <> value 8081
+    <> showDefault
+    <> help "Prometheus listening port" )
+  <*> strOption
+  ( long "target"
+    <> short 't'
+    <> metavar "TARGET"
```

Comment:
rename to `UPLOAD_TARGET` for compatibility? We use this in the systemd unit.

Reply:
it doesn't actually read from environment, so we'll have to change the unit anyways